src/osyris/io/sink.py

```diff
@@ -52,7 +52,12 @@ def initialize(self, meta, select):
 
         sink = Datagroup()
         for i, (key, unit) in enumerate(zip(key_list, unit_list)):
-            sink[key] = Array(values=sink_data[:, i] * unit.magnitude, unit=unit.units)
+            if len(sink_data.shape) == 1:
```

Comment:
Please have a look at #47 to see if that would fix your issue with sinks

Reply:
This seems to load the data correctly (and it's much simpler), but for some reason it gives an error when attempting to make a scatter plot of the sinks. I'll try to see what's causing this later this afternoon.

```---------------------------------------------------------------------------
DimensionalityError                       Traceback (most recent call last)
~/Desktop/osyris_work/test_workspace/line_integral_conv.py in <module>
     59              dx=dx, origin=center, direction="z", ax=ax[1])
     60 """
---> 61 plane = osyris.plane(#{"data": data["hydro"]["velocity"], 
     62                      #"mode": "lic", "length":length, "color":data["hydro"]["density"], "norm":"log", "cmap":"viridis"},
     63                      {"data": data["sink"]["xyz"], "mode": "scatter", "c": "white"},

~/Desktop/osyris_work/mmgonz_osyris/osyris/src/osyris/plot/plane.py in plane(direction, dx, dy, fname, title, plot, mode, norm, vmin, vmax, operation, origin, resolution, ax, *layers, **kwargs)
    302         # Add scatter layer
    303         if len(to_scatter) > 0:
--> 304             _add_scatter(to_scatter=to_scatter,
    305                          origin=origin,
    306                          dir_vecs=dir_vecs,

~/Desktop/osyris_work/mmgonz_osyris/osyris/src/osyris/plot/plane.py in _add_scatter(to_scatter, origin, dir_vecs, dx, dy, ax)
     28     dist1 = np.sum(xyz * dir_vecs[0], axis=1)
     29     global_selection = np.arange(len(to_scatter[0]["data"]))
---> 30     select = np.ravel(np.where(np.abs(dist1) <= radius))
     31     global_selection = global_selection[select]
     32     if len(select) > 0:

~/Desktop/osyris_work/mmgonz_osyris/osyris/src/osyris/core/array.py in __le__(self, other)
    324 
    325     def __le__(self, other):
--> 326         return _comparison_operator(self, other, "less_equal")
    327 
    328     def __gt__(self, other):

~/Desktop/osyris_work/mmgonz_osyris/osyris/src/osyris/core/array.py in _comparison_operator(lhs, rhs, op)
     11     func = getattr(np, op)
     12     if isinstance(rhs, Array):
---> 13         scale_r = rhs.unit.to(lhs._unit.units)
     14         return func(lhs._array, rhs._array * scale_r.magnitude)
     15     if isinstance(rhs, Quantity):

~/opt/anaconda3/lib/python3.8/site-packages/pint/quantity.py in to(self, other, *contexts, **ctx_kwargs)
    722         other = to_units_container(other, self._REGISTRY)
    723 
--> 724         magnitude = self._convert_magnitude_not_inplace(other, *contexts, **ctx_kwargs)
    725 
    726         return self.__class__(magnitude, other)

~/opt/anaconda3/lib/python3.8/site-packages/pint/quantity.py in _convert_magnitude_not_inplace(self, other, *contexts, **ctx_kwargs)
    671                 return self._REGISTRY.convert(self._magnitude, self._units, other)
    672 
--> 673         return self._REGISTRY.convert(self._magnitude, self._units, other)
    674 
    675     def _convert_magnitude(self, other, *contexts, **ctx_kwargs):

~/opt/anaconda3/lib/python3.8/site-packages/pint/registry.py in convert(self, value, src, dst, inplace)
   1001             return value
   1002 
-> 1003         return self._convert(value, src, dst, inplace)
   1004 
   1005     def _convert(self, value, src, dst, inplace=False, check_dimensionality=True):

~/opt/anaconda3/lib/python3.8/site-packages/pint/registry.py in _convert(self, value, src, dst, inplace)
   1915                 value, src = src._magnitude, src._units
   1916 
-> 1917         return super()._convert(value, src, dst, inplace)
   1918 
   1919     def _get_compatible_units(self, input_units, group_or_system):

~/opt/anaconda3/lib/python3.8/site-packages/pint/registry.py in _convert(self, value, src, dst, inplace)
   1516 
   1517         if not (src_offset_unit or dst_offset_unit):
-> 1518             return super()._convert(value, src, dst, inplace)
   1519 
   1520         src_dim = self._get_dimensionality(src)

~/opt/anaconda3/lib/python3.8/site-packages/pint/registry.py in _convert(self, value, src, dst, inplace, check_dimensionality)
   1034             # then the conversion cannot be performed.
   1035             if src_dim != dst_dim:
-> 1036                 raise DimensionalityError(src, dst, src_dim, dst_dim)
   1037 
   1038         # Here src and dst have only multiplicative units left. Thus we can

DimensionalityError: Cannot convert from 'astronomical_unit ** 2' ([length] ** 2) to 'astronomical_unit' ([length])```

Reply:
My mystake, I had accidentally multiplied dx twice by the "au" unit. Matthias's solution works fine.